### PR TITLE
Add hibernate command

### PR DESCRIPTION
**Pull request by gabrieljackson**
This new command hibernates installations which have received no
new posts within the specified number of days. This is accomplished
by performing post count metric queries against prometheus data for
each installation.

Fixes https://mattermost.atlassian.net/browse/MM-32132

### CODE REVIEW

**Review by jwilander — APPROVED** [review]
Looks good 👍 Just a couple comments

File: cmd/fleet-controller/hibernate.go [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+			if installation.State != cmodel.InstallationStateStable {
+				logger.Warnf("%s - Expected only stable installations (%s); skipping...", installation.ID, installation.State)
+				continue
+			}
+			if installation.APISecurityLock && !unlock {
+				logger.Warnf("%s - Installation is locked and hibernator is not set to perform unlocks; skipping...", installation.ID)
+				continue
+			}
+
+			time.Sleep(100 * time.Millisecond)
```
[the remark this answers is not in the document]
Comment (gabrieljackson) [reply cmd/fleet-controller/hibernate.go:88]: Yeah, definitely using the force a bit here, but that felt like a good starting point. I will add a comment.

File: cmd/fleet-controller/hibernate.go [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+
+	return shouldHibernate(rawMetrics), nil
+}
+
+func shouldHibernate(rawMetrics pmodel.Matrix) bool {
+	if len(rawMetrics) == 0 {
+		return false
+	}
+
+	for _, rawMetric := range rawMetrics {
```
[the remark this answers is not in the document]
Comment (gabrieljackson) [reply cmd/fleet-controller/hibernate.go:181]: Seems pretty obvious to me... ;)

Nah, it definitely could be explained a bit more.

**Review by stylianosrigas — APPROVED** [review]
LGTM, great work dude 👌

**Pull request merged**